app/models/user.rb

```diff
@@ -16,9 +18,9 @@ class User < ActiveRecord::Base
   attr_accessible *(ACCESSIBLE_ATTRIBUTES + [:admin]), :as => :admin
 
   validates_presence_of :username
-  validates_uniqueness_of :username
+  validates :username, uniqueness: { case_sensitive: false }
```

Comment:
I got confused, we actually want the comparison to be case-__sensitive__ so there can't be two users named `admin` and `Admin`. The option is a bit confusing `case_sensitive: false` makes the uniqueness constraint case insensitive and thus no allowing `admin` and `Admin`.